lib/google_crawler/search/search_keyword_worker.ex

```diff
@@ -1,4 +1,9 @@
 defmodule GoogleCrawler.SearchKeywordWorker do
```

Comment:
I am really not confident about this genserver 😂 

I think it's because

1. The state is quite complex I think 🤔 💭 - Now it stores a map of `%{task_ref -> {%Keyword{}, retry_count}}`.  So I don't know if it is hard to read or not

2. I am not sure about the error handling part. There are 2 cases now
  - When the task is failed ->  This handles by `handle_info({:DOWN}, ...)`
  - When something is wrong when it insert the record -> This also needs to retry, so I extracted the retry logic out to a new function. It looks a bit weird.

Main problem is I'm not sure if it should be like this or not. Is it the right way to do it or not. Probably I need to learn more about the Genserver 😵